File: 20200122/ironstick.test.js
const solution = (arrangement) =>
  [...arrangement.replace(/\(\)/g, '*')]
    .reduce(({ sticks, total }, bracket) => ({
      sticks: sticks + (bracket === '(' && 1) + (bracket === ')' && -1),
      total: total + (bracket === '*' && sticks) + (bracket === ')' && 1),
    }), { sticks: 0, total: 0 }).total;

test('solution', () => {
  expect(solution('()')).toBe(0);
  expect(solution('(()())')).toBe(3);
  expect(solution('()(((()())(())()))(())')).toBe(17);
});

File: 20200312/python/ironstick_test.py
def solution(arrangement):
    sticks = 0
    total = 0
    for c in arrangement.replace('()', '*'):
        if c == '(':
            sticks += 1
        if c == '*':
            total += sticks
        if c == ')':
            sticks -= 1
            total += 1
    return total


def test_solution():
    assert solution('(()())') == 3
    assert solution('()(((()())(())()))(())') == 17
